diff --git a/include/env_default.h b/include/env_default.h
index 86b639d3e2..b18244f674 100644
--- a/include/env_default.h
+++ b/include/env_default.h
@@ -9,6 +9,8 @@
 
 #include <env_callback.h>
 
+#include <env_resin.h>
+
 #ifdef DEFAULT_ENV_INSTANCE_EMBEDDED
 env_t environment __UBOOT_ENV_SECTION__(environment) = {
 	ENV_CRC,	/* CRC Sum */
@@ -21,6 +23,7 @@ static char default_environment[] = {
 #else
 const uchar default_environment[] = {
 #endif
+    RESIN_ENV
 #ifndef CONFIG_USE_DEFAULT_ENV_FILE
 #ifdef	CONFIG_ENV_CALLBACK_LIST_DEFAULT
 	ENV_CALLBACK_VAR "=" CONFIG_ENV_CALLBACK_LIST_DEFAULT "\0"
